autoload/lsp/lsp.vim

```diff
@@ -45,13 +45,41 @@ enddef
 
 # Returns the LSP server for the a specific filetype. Returns an empty dict if
 # the server is not found.
-def LspGetServer(ftype: string): dict<any>
-  return ftypeServerMap->get(ftype, {})
+def LspGetServer(bnr: number, ftype: string): dict<any>
+  if ftypeServerMap->has_key(ftype)
+    var lspservers = ftypeServerMap[ftype]
+
+    # The best language server for the current buffer
+    var bestMatch: dict<any> = {}
+    var bestScore: number = -1
+
+    var bufDir = bnr->bufname()->fnamemodify(':p:h')
+
+    for lspserver in lspservers
+      var score: number = 0
+      if !lspserver.rootSearchFiles->empty()
+        # The score is calculated by how deep the workspace root dir is, the
+        # deeper the better.
+        var path = util.FindNearestRootDir(bufDir, lspserver.rootSearchFiles)
+        score = path->strcharlen()
+      endif
```

Comment:
Maybe support should instead be added to run multiple language servers for a given filetype, and then another option should be added called `runIf` which takes a function with the current `bnr`:

    runIf: (bnr) => {
      return lsp#util#FindNearestRootDir(bnr->bufname()->fnamemodify(':p:h'), [
        'package.json'
      ])
    }